src/analysis/complex/phragmen_lindelof.lean

```diff
@@ -0,0 +1,579 @@
+      continuous_on_const subset_closure subset.rfl },
+  set g : ℕ → ℂ → E := λ n z, (exp z) ^ n • f z,
+  have hg : ∀ n z, ∥g n z∥ = (expR z.re) ^ n * ∥f z∥,
+  { intros n z, simp only [norm_smul, norm_eq_abs, complex.abs_pow, abs_exp] },
+  intros z hz,
+  suffices H : ∀ n : ℕ, ∥g n z∥ ≤ C,
+  { contrapose! H,
+    simp only [hg],
+    exact (((tendsto_pow_at_top_at_top_of_one_lt (real.one_lt_exp_iff.2 hz)).at_top_mul
+      (norm_pos_iff.2 H) tendsto_const_nhds).eventually (eventually_gt_at_top C)).exists },
+  intro n,
+  refine right_half_plane_of_tendsto_zero_on_real (differentiable_exp.pow.diff_cont_on_cl.smul hd)
+    _ _ (λ y, _) hz.le,
+  { rcases hexp with ⟨c, hc, B, hO⟩,
+    refine ⟨max c 1, max_lt hc one_lt_two, n + max B 0, is_O.of_norm_left _⟩,
+    simp only [hg],
+    refine ((is_O_refl (λ z : ℂ, expR z.re ^ n) _).mul hO.norm_left).trans (is_O.of_bound 1 _),
+    simp only [← real.exp_nat_mul, ← real.exp_add, real.norm_of_nonneg (real.exp_pos _).le,
+      real.exp_le_exp, add_mul, eventually_inf_principal, eventually_comap, one_mul],
+    filter_upwards [eventually_ge_at_top (1 : ℝ)] with r hr z hzr hre, subst r,
+    refine add_le_add (mul_le_mul_of_nonneg_left _ n.cast_nonneg) _,
+    { calc z.re ≤ abs z : re_le_abs _
+      ... = abs z ^ (1 : ℝ) : (real.rpow_one _).symm
+      ... ≤ abs z ^ (max c 1) : real.rpow_le_rpow_of_exponent_le hr (le_max_right _ _) },
+    { exact mul_le_mul (le_max_left _ _) (real.rpow_le_rpow_of_exponent_le hr (le_max_left _ _))
+        (real.rpow_nonneg_of_nonneg (abs_nonneg _) _) (le_max_right _ _) } },
+  { rw tendsto_zero_iff_norm_tendsto_zero, simp only [hg],
+    exact hre n },
+  { rw [hg, of_real_mul_re, I_re, mul_zero, real.exp_zero, one_pow, one_mul],
+    exact hC y }
+end
+
+/-- **Phragmen-Lindelöf principle** in the right half-plane. Let `f g : ℂ → E` be functions such
+that
+
+* `f` and `g` are differentiable in the open right half-plane and are continuous on its closure;
+* `∥f z∥` and `∥g z∥` are bounded from above by `A * exp(B * (abs z) ^ c)` on the open right
+  half-plane for some `c < 2`;
+* `∥f z∥` anre `∥g z∥` are bounded from above by constants on the imaginary axis;
+* `f x` and `g x`, `x : ℝ`, tend to zero superexponentially fast as `x → ∞`:
+  for any natural `n`, `exp (n * x) * ∥f x∥` and `exp (n * x) * ∥g x∥` tend to zero as `x → ∞`.
+
+Then `f` is equal to `g` on the closed right half-plane. -/
+lemma eq_on_right_half_plane_of_superexponential_decay {g : ℂ → E}
+  (hfd : diff_cont_on_cl ℂ f {z | 0 < z.re}) (hgd : diff_cont_on_cl ℂ g {z | 0 < z.re})
+  (hfexp : ∃ (c < (2 : ℝ)) B, is_O f (λ z, expR (B * (abs z) ^ c))
+    (comap abs at_top ⊓ 𝓟 {z | 0 < z.re}))
+  (hgexp : ∃ (c < (2 : ℝ)) B, is_O g (λ z, expR (B * (abs z) ^ c))
+    (comap abs at_top ⊓ 𝓟 {z | 0 < z.re}))
+  (hfre : superpolynomial_decay at_top expR (λ x, ∥f x∥))
+  (hgre : superpolynomial_decay at_top expR (λ x, ∥g x∥))
+  (hfim : ∃ C, ∀ x : ℝ, ∥f (x * I)∥ ≤ C) (hgim : ∃ C, ∀ x : ℝ, ∥g (x * I)∥ ≤ C) :
```

Comment:
Your assumptions imply that both `f` and `g` vanish on the closed right half-plane (just by applying `eq_zero_on_right_half_plane_of_superexponential_decay`, for which you have all the assumptions unless I am mistaken). So of course they are equal, but I am not sure it is worth stating this as a separate lemma. Unless you were thinking of a different statement, of course.